ambry-protocol/src/main/java/com/github/ambry/protocol/FileCopyGetMetaDataRequest.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2024 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.protocol;
+
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.utils.Utils;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+
+public class FileCopyGetMetaDataRequest extends RequestOrResponse{
+  private PartitionId partitionId;
+  private String hostName;
+  private static final short File_Metadata_Request_Version_V1 = 1;
+  private static final int HostName_Field_Size_In_Bytes = 4;
+
+  public FileCopyGetMetaDataRequest(short versionId, int correlationId, String clientId,
+      PartitionId partitionId, String hostName) {
+    super(RequestOrResponseType.FileCopyGetMetaDataRequest, versionId, correlationId, clientId);
+    if (partitionId == null || hostName.isEmpty()) {
+      throw new IllegalArgumentException("Partition and Host Name cannot be null");
```

Comment:
minor: We should have different error messages for both cases.

Reply:
Incorporated.